### ISSUE TITLE
URGENT: Replicate Physician Registration Page with Unbounce Implementation

### ISSUE THREAD

**Issue by sjohnson620** [nthopinion/covid19#88]
The new Unbounce landing page will be using the URL the existing Physician Registration page. We will need to replicate the registration fields on the Sign In page on the right hand side of the screen.
 
**Steps:**
1. User will select "Join the Community"
2. User will complete fields below on the right hand side of screen. When "Request Invite Now" is selected, user will be presented with "Registration Sent" popup.
(**URL**: https://www.covid19webapp.com/bIiOOIIqgwEXwUU3SaD0F9)

**Fields:**
- Email
- Role: (Dropdown)
     - Physician
     -Healthcare Provider
     -Other

-Select Country (Dropdown)

-Profile Link (Clinic)

-Medical Provider Number (eg. NPI Identifier)

-Be anonymous (checkbox)

**New Landing Page**
![image](https://user-images.githubusercontent.com/62316151/77990122-1e065f00-72e6-11ea-964f-a08a0a19ddb9.png)

**Current Sign/In Register**
![image](https://user-images.githubusercontent.com/62316151/77990133-2d85a800-72e6-11ea-859e-64c02c7f5649.png)

**Existing Registration Page**
![image](https://user-images.githubusercontent.com/62316151/77990096-0dee7f80-72e6-11ea-89a2-907cdc16f0cf.png)

**Registration Sent Popup**
![image](https://user-images.githubusercontent.com/62316151/78103678-237dab00-73b3-11ea-9d67-3d1b90292e01.png)

┆Issue is synchronized with this [Trello card](https://trello.com/c/QTIf6oJ6) by [Unito](https://www.unito.io/learn-more)

**Comment by shangardezi** [comment]
I posted on Slack too, commenting here for visibility. 
This requires changes to the registration form.  However, it does not look like this form is part of this repo. 
So a few questions:
Where can this form be changed to incorporate the new fields?
There's also SSO with Google, how should this be handled? Currently, a new user can register via this method.

**Comment by raynman1** [comment]
Hello @shangardezi,

Thanks for helping with this! Indeed, the account creation is done with Azure ADB2C. @sethuraj can work with you to clarify and add the necessary form fields I believe!

Ray

**Comment by nthopinion** [comment]
Hi all
The login dialog is through Azure ADb2C. We could add more fields there , but we also use federated google authentication.. so please let me know whether this workflow makes sense
see below soon....

**Comment by nthopinion** [comment]
We will maintain the registration form in our own database separately (move from unbounce to our database). Once signed up and when the suer sign in, the user email id is authenticated with the registration data . If email is not in force redirect to the registration form. Please let me know. I will create the nosqldb container for that.

**Comment by nthopinion** [comment]
alternate opton is to remove google authentication and user our own sign in. We can bring additional columns here

**Comment by Jennything** [comment]
Hi, can we please have the sign in and sign up button separate as per my screenshot below?
![Login](https://user-images.githubusercontent.com/56145322/78098915-397c7300-7395-11ea-945b-ea25c143db5d.png)

**Comment by sjohnson620** [comment]
@Jennything Just updated the description based on my conversation with Ray and Jay. We will need to redesign login as a whole in the future- there are some downsides to what is being used now for logging into the site. It'll be a much bigger discussion.

Right now priority is to ensure there is an option for registering using existing form as we are implementing the new Unbounce page soon. Since this page will take place of the registration page, just need something on these for now so we don't have physicians unable to sign up.

@noahub @shangardezi  Whoever is able to pick this up first, please let the other person know so work isn't duplicated. 

@sethuraj has details on the container he can share with you.

**Comment by UnitoSyncBot** [comment]
➤ jul commented:

sethurajjay still need access to the Azure tool. moved this to "DOING" with Christina to better reflect the state of this task, but I'm not proceeding because I'm blocked

**Comment by UnitoSyncBot** [comment]
➤ jul commented:

sethurajjay rogerng7 rayng3 anishasunkerneni jennykim137 aviskarkc10 
from the lack of response, it looks like we should forget about the Azure tool and focus on implementing our own form and authentication?

**Comment by UnitoSyncBot** [comment]
➤ jul commented:

sethurajjay rogerng7 rayng3 anishasunkerneni jennykim137 aviskarkc10

hi:) could someone clarify today/over the weekend the way to proceed so that I (or someone else) can get back to this on Monday with no blockers?

currently, there's authentication set up using msal (https://www.npmjs.com/package/msal) that works with the Azure AD B2C service https://docs.microsoft.com/en-us/azure/active-directory-b2c/overview#identity-providers). 

I supposed that the thing to do then would be to use that service and add the extra fields we need, since the authentication and form are already in place. so I had asked for a way into the service to take a look around and see how it works and if achieving the goals won't be too complex using that, but didn't get access. I marked it as a blocker.  -> sethurajjay aviskarkc10 
here I'll note that I'm not familiar with the Azure service, but I'm ready to see how it works once given the access. then again, if it's better for someone else to take care of that, I'll move on to other tasks.

if it's not the way to go, then it seems like this feature consists of rewriting the registration/sign in completely, without using those tools. that means not just adding a form to send more data (that which is crucial to the registration process and verifying physicians) but also setting up the authentication from scratch. am I thinking correctly?

if this is the case - if we're to drop the current solution whatsoever and build our own form - it would be great to have at least a simple mockup that would show the placement of the component and the new flow (since in this case we're dropping the Azure tool flow and popup and going for something else) -> anishasunkerneni jennykim137

once it's clear which way we need to go, we can proceed further.

thanks!

/cc cnagadya

**Comment by UnitoSyncBot** [comment]
➤ jul commented:

anishasunkerneni I added designs to the list of blockers for this task

**Comment by UnitoSyncBot** [comment]
➤ Emem commented:

NthOpinionRedirectTest-01.mp4 ( https://trello-attachments.s3.amazonaws.com/5e66fa1975537834d6d47c45/5ea25560fc888423fbd08268/af350c057d825d61d872661bab5da817/NthOpinionRedirectTest-01.mp4 )

From the video you can see that the redirection is working. But I need to find out why the user is not signed in

**Comment by UnitoSyncBot** [comment]
➤ jul commented:

emem22

  1. I'd link to the new site directly from the qna page, right now we go to the old log in page and then when we click the button we're taken to the new page, which is weird and not great UX.
  2. when the page is loading, there's a flash of it with no styles applied, have you noticed that? I think this should be fixed. do you know why this is happening?
  3. I noticed that when I'm logged in and on the dashboard and refresh the page there's a brief flash of the old page
  4. the title of the new log in page is "Loading..." - is there a way to change that? see screenshot ( https://trello-attachments.s3.amazonaws.com/5e66fa1975537834d6d47c45/5ea25560fc888423fbd08268/d33b2a72cde4a133f384048dfd9a3c16/Screenshot_2020-06-23_at_18.03.20.png )

I'd also tweak the styling a bit, but it's probably not worth it right now, since the redesign is coming.

BTW I was just able to break the page when testing locally to double check point 1 and 3

**Comment by UnitoSyncBot** [comment]
➤ jul commented:

emem22 I fixed the app by removing cookies, so I'm guessing there's something wrong with storing the session?

**Comment by UnitoSyncBot** [comment]
➤ jul commented:

sethurajjay were you able to test this? did you run into the issues I've encountered?

**Comment by UnitoSyncBot** [comment]
➤ jul commented:

sethurajjay emem22

registration on production looks like this ( https://trello-attachments.s3.amazonaws.com/5e66fa1975537834d6d47c45/5ea25560fc888423fbd08268/d69bbbcb564f728313f2e79c0ac0e21d/Screenshot_2020-06-24_at_00.02.05.png ) at the moment - the content of the popup is now our new registration page. so until this change is merged we got the broken looking registration.

still, we need to test this extensively and make sure everything is smooth.

if there's a way to change the settings in the azure tool so that we have the normal popup on production we should do it.

**Comment by UnitoSyncBot** [comment]
➤ jul commented:

emem22 just tried logging in on staging, got redirected to the old log in page: https://ntozwu-pd-covid19-app.azurewebsites.net/bIiOOIIqgwEXwUU3SaD0F9#

checked again, same.
I think the flow needs some more care.

**Comment by UnitoSyncBot** [comment]
➤ jul commented:

a bit of a mixup happened here, because this issue got merged to master, so it seems like it was accepted, and yet it's still in this column and emem22 is taking care of some remaining issues related to it. not an optimal situation, because the issue got closed in GH since it was merged to master, and yet it's still not completed. 
but for now let's forget about the GH flow, which needs optimising too.

a couple of things we can improve here:

  * if a task was accepted/merged it's supposed to be moved forward (and since aviskarkc10 and sethurajjay are those who take care of that, it's their responsibility)
  * we have to make sure that features are as well tested as possible before they're accepted
  * if bugs are discovered, new cards representing them can be added to TODO column. this way it's more visible what's going on, especially if the main feature card is already in one of the last columns and a PR for it was already merged to master. in such a case the feature can be marked as BLOCKED as well and unblocked after bug fixes are accepted.

/cc aviskarkc10 sethurajjay cnagadya emem22

**Comment by UnitoSyncBot** [comment]
➤ jul commented:

emem22 aviskarkc10 sethurajjay can we extract the remaining issue to another card? it seems to me that it isn't a no-deal bug. then this task could be tested apart from that and if everything else works we could deploy to unblock other cards. :pray:
